hoomd/md/pytest/test_lj_equation_of_state.py

```diff
@@ -0,0 +1,127 @@
+    if method_name == 'NVT':
+        method = hoomd.md.methods.NVT(filter=hoomd.filter.All(),
+                                      kT=T_star,
+                                      tau=0.1)
+    elif method_name == 'Langevin':
+        method = hoomd.md.methods.Langevin(filter=hoomd.filter.All(), kT=T_star)
+    elif method_name == 'NPT':
+        method = hoomd.md.methods.NPT(filter=hoomd.filter.All(),
+                                      kT=T_star,
+                                      tau=0.1,
+                                      S=mean_P_ref,
+                                      tauS=0.5,
+                                      couple='xyz')
+    integrator.methods.append(method)
+    sim.operations.integrator = integrator
+
+    # equilibrate the simulation
+    sim.run(0)
+    sim.state.thermalize_particle_momenta(filter=hoomd.filter.All(), kT=T_star)
+    if method_name == 'NVT':
+        method.thermalize_thermostat_dof()
+    elif method_name == 'NPT':
+        method.thermalize_thermostat_and_barostat_dof()
+    sim.run(equilibration_steps)
+
+    # log energy and pressure
+    thermo = hoomd.md.compute.ThermodynamicQuantities(filter=hoomd.filter.All())
+    sim.operations.computes.append(thermo)
+    energy_log = hoomd.conftest.ListWriter(thermo, 'potential_energy')
+    pressure_log = hoomd.conftest.ListWriter(thermo, 'pressure')
+    volume_log = hoomd.conftest.ListWriter(thermo, 'volume')
+    log_trigger = hoomd.trigger.Periodic(log_period)
+    sim.operations.writers.append(
+        hoomd.write.CustomWriter(action=energy_log, trigger=log_trigger))
+    sim.operations.writers.append(
+        hoomd.write.CustomWriter(action=pressure_log, trigger=log_trigger))
+    sim.operations.writers.append(
+        hoomd.write.CustomWriter(action=volume_log, trigger=log_trigger))
+
+    sim.always_compute_pressure = True
+    sim.run(run_steps)
+
+    # compute the average and error
+    energy = hoomd.conftest.BlockAverage(numpy.array(energy_log.data) / N)
+    pressure = hoomd.conftest.BlockAverage(pressure_log.data)
+    rho = hoomd.conftest.BlockAverage(N / numpy.array(volume_log.data))
+
+    # Useful information to know when the test fails
+    print('U_ref = ', mean_U_ref, '+/-', sigma_U_ref)
+    print('U = ', energy.mean, '+/-', energy.standard_deviation, '(',
+          energy.relative_error * 100, '%)')
+    print('P_ref = ', mean_P_ref, '+/-', sigma_P_ref)
+    print('P = ', pressure.mean, '+/-', pressure.standard_deviation,
+          pressure.standard_deviation, '(', pressure.relative_error * 100, '%)')
+    print('rho = ', rho.mean, '+/-', rho.standard_deviation)
+
+    print(f'Statepoint entry: {T_star:0.4}, {rho_star:0.4}, '
+          f'{energy.mean:0.5}, {energy.standard_deviation:0.4}, '
+          f'{pressure.mean:0.5}, {pressure.standard_deviation:0.4}')
+
+    energy.assert_close(mean_U_ref, sigma_U_ref)
+
+    # use larger tolerances for pressure and density as these have larger
+    # fluctuations
```

Comment:
It looks like the reference deviation for density is 0 and this would mean a lower tolerance?

Reply:
The tolerances are `z` and `max_relative_error`. The 0 is correct - the reference run for this state point was an NVT simulation where density is constant.

Reply:
I wasn't arguing for correctness, but the density check has the default `max_relative_error` and `z`, and the zero doesn't change the tolerance terms, but the comment mentions it does, and the same for the pressure actually since you provide a `sigma_P_ref` and use that to create the new Gaussian `reference + computed` the provided std isn't actually an increased tolerance.

I did in the first pass think the `0` was related to the tolerance via the comment, but now I know it is just defining the reference distribution.

```suggestion
```

Reply:
You are right, the comment should be removed.

At various stages of the testing, I did modify the tolerance with `z` and `max_relative_error` for these two checks separately. But I do not in the current code.